packages/workspace/src/core/hasher/git-hasher.spec.ts

```diff
@@ -1,6 +1,47 @@
 import { dirSync } from 'tmp';
 import { mkdirSync, removeSync } from 'fs-extra';
-import { execSync } from 'child_process';
+import {
+  execSync,
+  SpawnSyncOptionsWithBufferEncoding,
+  SpawnSyncReturns,
+} from 'child_process';
+
+jest.doMock('child_process', () => {
+  return {
+    // @ts-ignore
+    ...jest.requireActual('child_process'),
+    spawnSync: jest
+      .fn()
+      .mockImplementation(
+        (
+          command: string,
+          args?: ReadonlyArray<string>,
+          options?: SpawnSyncOptionsWithBufferEncoding
+        ): SpawnSyncReturns<string> => {
+          // Conditionally mock spawnSync for a specific unit test using __USE_MOCKED_SPAWN_SYNC__
```

Comment:
It felt like a good idea to come up with some way to test this to protect against regressions and this was the best I could come up with quickly to have a conditional mock of an implementation detail

Reply:
Maybe extracting the `parseGitStatus` function from the `git-hasher` to test it independently would help, but that's not a game changer imo.